src/OptimisticTokenVotingPlugin.sol

```diff
@@ -219,6 +227,11 @@ contract OptimisticTokenVotingPlugin is
         if ((_block.proposedAt + governanceSettings.l2InactivityPeriod) < block.timestamp) return false;
 
         return true;
+
+
+        // new impl suggested by Daniel W
+        // (,, uint64 lastBlockVerifiedTimestamp) = taikoL1.getLastVerifiedBlock();
+        // return _block.proposedAt + governanceSettings.l2InactivityPeriod >= block.timestamp;
```

Comment:
It appears as  `_block.proposedAt` should also be removed?

Would this implementation look correct to you?

```solidity
        // Actively disabled L2 voting?
        if (governanceSettings.skipL2) return false;
        // Is the L1 bridge paused?
        else if (taikoL1.paused()) return false;

        uint64 _id = taikoL1.slotB().numBlocks;
        // No L2 blocks yet
        if (_id == 0) return false;

        (,, uint64 lastBlockVerifiedTimestamp) = taikoL1.getLastVerifiedBlock();
        return (lastBlockVerifiedTimestamp + governanceSettings.l2InactivityPeriod) >= block.timestamp;
```

Reply:
We should also upgrade to use version 1.8.0 of the protocol. Is this correct?

Reply:
Yes please, then later we will release a 1.9.0 for our Ontake hardfork.

Reply:
I see that the current signature on 1.8.0 is `function getLastVerifiedBlock() external view returns (uint64 blockId_, bytes32 blockHash_, bytes32 stateRoot_)`

Whereas the example code is retrieving `(,, uint64 lastBlockVerifiedTimestamp)`, which doesn't seem to match.
Is this the change that will be landing on 1.9.0?

Reply:
Yes, it will be part of 1.9.0

Reply:
See https://github.com/aragon/taiko-contracts/pull/39